xdsl/irdl.py

```diff
@@ -219,7 +219,9 @@ def irdl_to_attr_constraint(
 
     # Attribute class case
     # This is a coercion for an `BaseAttr`.
-    if isclass(irdl) and issubclass(irdl, Attribute):
+    if isclass(irdl) \
+            and not isinstance(irdl, GenericAlias) \
```

Comment:
What's the impact of this additional condition, and its link to `ArrayAttr` being `Iterable`?

Reply:
Not exactly sure, it seems that having generic + multiple inheritance (possibly from `Iterable` also not being a class in the classical sense) causes the type to become a `GenericAlias`, which satisfies `isclass`, but cannot be passed to `issubclass`... This fixes it for now, but @math-fehr might have a better solution down the road.

Reply:
I don't know this corner of IRDL well, so I'm not sure, but it reads to me like it's plainly ignoring some constraints here. Maybe I'm just missing something?